src/cli.py

```diff
@@ -0,0 +1,71 @@
+
+    if not codeQL_exists and (whichtwo == "CQLvCNL" or whichtwo == "CLJvCQL" or whichtwo == "all"):
+        raise FileNotFoundError(f"\033[91mCodeQL source does not exist: \033[0m{codeQL_directory}")
+    
+    if not clojure_exists and (whichtwo == "CLJvCNL" or whichtwo == "CLJvCQL" or whichtwo == "all"):
+        raise FileNotFoundError(f"\033[91mClojure source does not exist: \033[0m{clojure_directory}")
+    
+    if not coccinelle_exists and (whichtwo == "CQLvCNL" or whichtwo == "CLJvCNL" or whichtwo == "all"):
+        raise FileNotFoundError(f"\033[91mCoccinelle source does not exist: \033[0m{coccinelle_directory}")
+    
+    if not whichtwo:
+        if codeQL_exists and clojure_exists and coccinelle_exists:
+            whichtwo = "all"
+        elif codeQL_exists and clojure_exists:
+            whichtwo = "CLJvCQL"
+        elif coccinelle_exists and clojure_exists:
+            whichtwo = "CLJvCNL"
+        elif codeQL_exists and coccinelle_exists:
+            whichtwo = "CQLvCNL"
+        else:
+            print(f"\033[91mWARNING: \033[0mOnly one or fewer input source for comparsion, and the output could be meaningless.")
+            sys.exit(0)
+        print(f"whichtwo:", whichtwo)
+
+    if output_path:
+        output_path = Path(output_path)
+        if not output_path.is_dir():        
+            output_path.mkdir(parents = True, exist_ok = True)
+            print(f"\033[34mATTENTION: \033[0mnew folder created at: {output_path}")
+
+    if not output_path:
+        output_path = Path(Path.cwd(), "output")
+
+        if not output_path.is_dir():        
+            output_path.mkdir(parents = True, exist_ok = True)
+            print(f"\033[34mATTENTION: \033[0mDefault output folder created at:", output_path)
+
+    return output_path, codeQL_directory, clojure_directory, coccinelle_directory, whichtwo
+
+@click.command()
+@click.option('--input', type = str, required = True, help = 'Directory containing three folders of source data.')
+@click.option('--output', type = str, help = 'Directory to store output files. The input is optional.')
+@click.option('--codeql_relative', type = str, required = True, help = 'Relative path to truncate from file names in the CodeQL data.')
+@click.option('--clojure_relative', type = str, help = 'Relative path to truncate from file names in the Clojure data.')
+@click.option('--coccinelle_relative', type = str, required = True, help = 'Relative path to truncate from file names in the Coccinelle data.')
+@click.option('--whichtwo', type=click.Choice(['CLJvCQL', 'CLJvCNL', 'CQLvCNL', 'all']), help = "This option selects two, or all, of the three tools to be compared. Four possible inputs are supported: 'CLJvCQL', 'CLJvCNL', 'CQLvCNL', and 'all'. If omitted, the program will assign a value to this option by checking input folder availiability.")
+def cli(input, output, codeql_relative, clojure_relative, coccinelle_relative, whichtwo):
```

Comment:
We should use a more intuitive name. It's really unusual for a command to be called `cli`